samples/JobsAgent/JobsAgent.cpp

```diff
@@ -0,0 +1,291 @@
+
+            p_iot_client_ = std::shared_ptr<MqttClient>(MqttClient::Create(p_network_connection_,
+                                                                           ConfigCommon::mqtt_command_timeout_,
+                                                                           p_disconnect_handler, nullptr,
+                                                                           p_reconnect_handler, nullptr,
+                                                                           p_resubscribe_handler, nullptr));
+            if (nullptr == p_iot_client_) {
+                return ResponseCode::FAILURE;
+            }
+
+            util::String client_id_tagged = ConfigCommon::base_client_id_;
+            client_id_tagged.append("_jobs_agent_");
+            client_id_tagged.append(std::to_string(rand()));
+            std::unique_ptr<Utf8String> client_id = Utf8String::Create(client_id_tagged);
+
+            rc = p_iot_client_->Connect(ConfigCommon::mqtt_command_timeout_, ConfigCommon::is_clean_session_,
+                                        mqtt::Version::MQTT_3_1_1, ConfigCommon::keep_alive_timeout_secs_,
+                                        std::move(client_id), nullptr, nullptr, nullptr);
+            if (ResponseCode::MQTT_CONNACK_CONNECTION_ACCEPTED != rc) {
+                return rc;
+            }
+
+            StartInstalledPackages();
+
+            p_jobs_ = Jobs::Create(p_iot_client_, mqtt::QoS::QOS1, ConfigCommon::thing_name_, client_id_tagged);
+
+            rc = Subscribe();
+            if (ResponseCode::SUCCESS != rc) {
+                AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT, "Subscribe failed. %s", ResponseHelper::ToString(rc).c_str());
+            } else {
+                rc = p_jobs_->SendJobsQuery(Jobs::JOB_GET_PENDING_TOPIC);
+
+                if (ResponseCode::SUCCESS == rc) {
+                    rc = p_jobs_->SendJobsQuery(Jobs::JOB_DESCRIBE_TOPIC, "$next");
+                }
+
+                if (ResponseCode::SUCCESS != rc) {
+                    AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT, "SendJobsQuery failed. %s",
+                                  ResponseHelper::ToString(rc).c_str());
+                    p_iot_client_->Disconnect(ConfigCommon::mqtt_command_timeout_);
+                }
+            }
+
+            // Wait for job processing to complete
+            done_ = false;
+            while (!done_) {
+                std::this_thread::sleep_for(std::chrono::seconds(10));
+            }
+
+            rc = p_iot_client_->Disconnect(ConfigCommon::mqtt_command_timeout_);
+            if (ResponseCode::SUCCESS != rc) {
+                AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT, "Disconnect failed. %s", ResponseHelper::ToString(rc).c_str());
+            }
+
+            std::cout << "Exiting Sample!!!!" << std::endl;
+            return ResponseCode::SUCCESS;
+        }
+    }
+}
+
+int main(int argc, char **argv) {
+    curl_global_init(CURL_GLOBAL_DEFAULT);
```

Comment:
Is this supposed to work on windows? If this is running on the CPP sdk, doesn't it run on winhttp?

Reply:
This is compatible with with Windows. Adding http client support to the IoT Device CPP SDK may be worth exploring, but probably also out of scope of this sample at this time.